scripts/trigger_ironside_ipc_request.py

```diff
@@ -0,0 +1,271 @@
+                if (perf_counter() - start) > timeout_s:
+                    break
+
+            call_buffer_data = bytes(
+                self._jlink.memory_read8(self._buffer_addr, c.sizeof(CallBuffer))
+            )
+            call_buffer = CallBuffer.from_buffer_copy(call_buffer_data)
+            has_data = call_buffer.status not in (
+                IronSideCallStatus.IDLE.value,
+                IronSideCallStatus.REQ.value,
+            )
+            if (perf_counter() - start) > timeout_s:
+                break
+
+        if not has_event:
+            raise IronSideCallTimeoutError(
+                f"timed out waiting for RX event at 0x{self._rx_event_addr:09_x}"
+            )
+        if not has_data:
+            raise IronSideCallTimeoutError("timed out waiting for response status")
+
+        assert call_buffer is not None
+
+        try:
+            rsp_status = call_buffer.status_enum
+        except ValueError:
+            raise IronSideCallUnknownStatusError(call_buffer.status)
+
+        if rsp_status != IronSideCallStatus.SUCCESS:
+            raise IronSideCallResponseError(rsp_status)
+
+        return [c.c_uint32(a) for a in call_buffer.args]
+
+
+class IronSideIpcTrigger:
+    """Class for triggering IronSide IPC requests"""
+
+    def __init__(self, jlink_serial: int):
+        self.jlink = pylink.JLink()
+        self.jlink.open(jlink_serial)
+        self.jlink.set_tif(pylink.enums.JLinkInterfaces.SWD)
+        self.jlink.set_speed(4000)
+        self.jlink.coresight_configure()
+        self.jlink.connect("Cortex-M33")
+
+        self.ironside_call = IronSideCall(
+            self.jlink,
+            tx_task_addr=0x5F09_9030, # CPUSEC.BELLBOARD.TASK[12]
+            rx_event_addr=0x5F09_A100, # CPUAPP.BELLBOARD.EVENT[0]
+            buffer_addr=0x2F88_FB80, # Static location of IPC buffer
+        )
+
+    def trigger_and_read_response(self) -> List[c.c_uint32]:
+        """
+        Trigger an IronSide IPC request and read the response.
+
+        Returns:
+            List of response arguments
+        """
+
+        self.ironside_call.trigger_request()
+        return self.ironside_call.read_response()
+
+    def close(self):
+        """Close the JLink connection"""
+        self.jlink.close()
+
+
+def program_hex_data(trigger: IronSideIpcTrigger, hex_data: str) -> None:
+    """
+    Program Intel HEX data to the device using the JLink connection.
+
+    Args:
+        trigger: IronSideIpcTrigger instance with active JLink connection
+        hex_data: Intel HEX format data as a string
+
+    Raises:
+        SystemExit: If Intel HEX data is empty or invalid
+    """
+    # Parse Intel HEX data
+    ih = IntelHex()
+    hex_io = io.StringIO(hex_data)
+    ih.loadhex(hex_io)
+
+    if len(ih) == 0:
+        print("Error: Intel HEX data is empty", file=sys.stderr)
+        sys.exit(1)
+
+    print("Halting CPU before flashing...")
+    trigger.jlink.halt()
```

Comment:
Note that we will also need to handle the DCACHE if the firmware has been running. If we are not planning on adding that handling now, could we maybe state somewhere that you should do `nrfutil device reset --reset-kind RESET_VIA_SECDOM` or equivalent before using this?